plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/commands/SQLCommandSet.java

```diff
@@ -32,18 +36,36 @@ public class SQLCommandSet implements SQLControlCommandHandler {
     @Override
     public boolean handleCommand(SQLControlCommand command, SQLScriptContext scriptContext) throws DBException {
         String parameter = command.getParameter();
-        int divPos = parameter.indexOf('=');
+        String varName = tryParseVariableName(scriptContext, parameter, 0);
+        if (varName == null) {
+            throw new DBCException("Missing variable name. Expected syntax:\n@set varName = value or expression");
+        }
+        int divPos = parameter.indexOf('=', varName.length());
         if (divPos == -1) {
             throw new DBCException("Bad set syntax. Expected syntax:\n@set varName = value or expression");
         }
-        String varName = parameter.substring(0, divPos).trim();
         String varValue = parameter.substring(divPos + 1).trim();
         varValue = GeneralUtils.replaceVariables(varValue, name -> CommonUtils.toString(scriptContext.getVariable(name)));
-        scriptContext.setVariable(varName, varValue);
+        scriptContext.setVariable(varName.trim(), varValue);
 
         return true;
     }
-
-    ;
-
+    
+    private String tryParseVariableName(SQLScriptContext ctx, String text, int start) {
+        while (start < text.length() && Character.isWhitespace(text.charAt(start))) {
+            start++;
+        }
+        SQLDialect sqlDialect = ctx.getExecutionContext().getDataSource().getSQLDialect();
+        int end = ScriptParameterRule.tryConsumeParameterName(sqlDialect, text, start);
+        if (end > 0) {
+            String rawParamName = text.substring(start, end);
+            if (end > 0 && sqlDialect.isQuotedIdentifier(rawParamName)) {
```

Comment:
The `end > 0` expression is always true

```suggestion
            if (sqlDialect.isQuotedIdentifier(rawParamName)) {
```